moduleframework/common.py

```diff
@@ -24,23 +25,46 @@
 Custom configuration and debugging library.
 """
 
-from __future__ import print_function
-
 import netifaces
 import socket
 import os
 import urllib
 import yaml
 import subprocess
 import copy
-import sys
 import random
 import string
 import requests
 import warnings
 import ast
-from avocado.utils import process
-from moduleframework.mtfexceptions import ModuleFrameworkException, ConfigExc, CmdExc, DefaultConfigExc
+import glob
+import avocado.utils
+import mtfexceptions
+import core
+
+
+class MTFConfParser(dict):
+    configdir = "mtf.conf.d"
+    source_cfg = os.path.join(*(["/"] + core.DATADIR + [configdir]))
```

Comment:
@jscotka What do you want to do with this command?

~~~
source_cfg = os.path.join(*(["/"] + ['etc']  + ["yum"]))
print source_cfg

prints /etc/yum
~~~

Why not only os.path.join(core.DATADIR, configdir)?
core.DATADIR could be 
~~~
core.DATADIR = "/usr/share/moduleframework"
~~~

Reply:
because threre is by intention missing leading ``/`` so that it created relative path, instead of ``/usr/share/moduleframework``, and  also it is caused by that this syntax was used in ``setup.py``  for this directory. So I did not want to change it in setup.py, just put it out of hardcoded value in setup.py